src/handler/events/observableHandling.ts

```diff
@@ -64,3 +68,55 @@ export function errTap<T extends Module>(cb: (err: SernError) => void) {
             });
         });
 }
+
+export function mod$<T extends CommandType>(module: CommandModule | undefined, cmdTy: T) {
+    return of(module).pipe(filterCorrectModule(cmdTy));
+}
+//POG
+export function isOneOfCorrectModules<T extends readonly CommandType[]>(...inputs: [...T]) {
+    return (src: Observable<CommandModule | undefined>) => {
+        return new Observable<CommandModuleDefs[T[number]]>(subscriber => {
+            return src.subscribe({
+                next(mod) {
+                    if (mod === undefined) {
+                        return throwError(() => SernError.UndefinedModule);
+                    }
+                    if (inputs.some(type => (mod.type & type) !== 0)) {
+                        subscriber.next(mod as CommandModuleDefs[T[number]]);
+                    } else {
+                        return throwError(() => SernError.MismatchModule);
+                    }
+                },
+                error: e => subscriber.error(e),
+                complete: () => subscriber.complete(),
+            });
+        });
+    };
+}
+
+export function executeModule(
+    wrapper: Wrapper,
+    payload: {
+        mod: CommandModule;
+        execute: () => unknown;
+        res: Result<void, void>[];
+    },
+) {
+    if (payload.res.every(el => el.ok)) {
+        return from(payload.execute() as Promise<unknown>).pipe(
```

Comment:
why need to cast it? what does it infer if not casted?

Reply:
it is inferred as unknown, but good point. i dont think casting is needed here

Reply:
nvm, its needed 
![image](https://user-images.githubusercontent.com/76754747/179337997-4a81c435-f04f-4b1d-b556-2c0814b58ebe.png)

Reply:
alright